cmd/run/run.go

```diff
@@ -867,11 +872,19 @@ func (s *ServerContext) Run(ctx context.Context, config *Config) error {
 			runtime.WithHealthzEndpoint(healthv1pb.NewHealthClient(conn)),
 			runtime.WithOutgoingHeaderMatcher(func(s string) (string, bool) { return s, true }),
 		}
+
 		mux := runtime.NewServeMux(muxOpts...)
 		if err := openfgav1.RegisterOpenFGAServiceHandler(ctx, mux, conn); err != nil {
 			return err
 		}
 
+		handler := http.Handler(mux)
+		if config.Trace.Enabled {
+			// if tracig is enabled wrap the handler with otel to extract the
+			// trace context from incoming requests
+			handler = otelhttp.NewHandler(mux, "proxy")
```

Comment:
The `proxy` intermediary is quite confusing. If we could annotate the endpoint handler with the method itself then it would make sense to me (e.g. the grpc method), but the operation 'proxy' doesn't mean anything to me.

I wonder if it makes more sense to just create a lightweight middleware that extracts the parent span from the incoming HTTP request context (if it exists) and inject it if it does. That would solve the same problem, remove the extra nested span with an odd name, and reduce one additional unecessary dependency.

What do you think?

Reply:
Yes, good idea. Adjusted the code with tiny wrapper, please check if that is what you meant.

Reply:
Looks better. Thanks!